### ISSUE TITLE
version 4.0.1 re2rust default generate C code

### ISSUE THREAD

**Issue by marryjianjian** [skvadrik/re2c#515]
Thanks for amazing tool.
Well, `re2rust` should be worked as `re2c --lang rust`. But recently I upgraded my re2c to version 4.0.1(my system is Manjaro, PKGBUILD file may same as Arch). ALL `re2rust/re2go/re2py/...` just act like `re2c`, they all generate C code files.

### version
![image](https://github.com/user-attachments/assets/fd7988eb-9078-434c-8763-79b166af42e5)

### my test
```bash
$ cat t.l 
fn lex(yyinput: &[u8]) -> bool {
    let mut yycursor = 0;
    /*!re2c
        re2c:YYCTYPE = u8;
        re2c:yyfill:enable = 0;

        [1-9][0-9]* { return true; }
        *           { return false; }
    */
}

fn main() {
    assert!(lex(b"1234\0"));
}
```
![image](https://github.com/user-attachments/assets/086e0049-0bb1-4616-b924-9b17452b7d63)
![image](https://github.com/user-attachments/assets/9da39eaa-760d-4daa-a4ed-a45b08f4544f)

The problem is re2xx always generating C code, but re2xx --lang xx works fine.

**Comment by skvadrik** [comment]
Thanks for reporting!

I can confirm there is a problem, and it is scoped to the CMake build system. Syntax files are built into an object library in order to ensure they build early, and this object library does not get the same compile definitions as the main binary. I will fix it shortly and release 4.0.2 with the update.

Autotools build system works fine.

**Comment by marryjianjian** [comment]
I'm glad it is helpful.:)

**Comment by skvadrik** [comment]
Fixed in re2c-4.0.2.
